lib/searchName.ml

```diff
@@ -130,25 +123,22 @@ type search_type =
   | DefaultSurname
 
 let search_for_fn_or_pn conf base fn pl =
+  let fn_l = cut_words fn in
   List.fold_left
     (fun pl p ->
       if search_reject_p conf base p then pl
       else
-        let fn1 =
-          get_first_name p |> sou base |> Name.lower |> Name.abbrev |> cut_words
-        in
-        let fn2 =
-          get_public_name p |> sou base |> Name.lower |> Name.abbrev
-          |> cut_words
-        in
-        let fnl = cut_words fn in
+        let fn1_l = get_first_name p |> sou base |> split_normalize in
+        let fn2_l = get_public_name p |> sou base |> split_normalize in
         let exact = false in
         (* TODO manage exact options according to mode *)
         if
-          if exact then fnl = fn1 || fnl = fn2
+          if exact then fn_l = fn1_l || fn_l = fn2_l
           else
-            List.fold_left (fun res fn -> res || List.mem fn fn1) false fnl
-            || List.fold_left (fun res fn -> res || List.mem fn fn2) false fnl
+            List.fold_left (fun res fn -> res || List.mem fn fn1_l) false fn_l
+            || List.fold_left
+                 (fun res fn -> res || List.mem fn fn2_l)
+                 false fn_l
```

Comment:
```ocaml
let tbl = Hashtbl.create 512 in
List.iter (fun k -> Hashtabl.add tbl k ()) fn1_l;
List.iter (fun k -> Hashtabl.add tbl k ()) fn2_l;
List.exists (Hashtbl.mem tbl) fn_l
```

To avoid being quadratic...

Reply:
I believe here the list is just small a list of words, we split the name we search for on whitespace

Reply:
Well, using `List.exists` instead of `List.fold_left (...) false l` is still easier to understand..